src/core/Database.cpp

```diff
@@ -296,7 +298,7 @@ bool Database::performSave(const QString& filePath, QString* error, bool atomic,
             }
```

Comment:
Wrong backup logic here? This backs up the *new* version *after* saving. Option says "backup *before* saving". Lines 316-318 use the correct logic.

I would suggest that the backup logic should be independent of the saving logic: back up first before line 286, then proceed to decide how to save.

Reply:
Indeed. I guess the logic here is that a backup will only be done after a successful write, whereas failed writes will leave the old backup in place. But the behaviour should be consistent across all save types.
Backing up first could involve some risk, but on the other hand, it would prevent data loss in cases of successful but corrupted writes. I guess I prefer backing up first with a failsafe in case the backup fails.

Reply:
It is definitely cleaner to just do a single backup call before doing any actual save operations. In the rare case where the backup fails, your save operation will also fail anyway.